sycl/ReleaseNotes.md

```diff
@@ -1,3 +1,254 @@
+- Added `host_task` in `sycl::target` enum class [a8401952]
+- Added support of `sycl::marray` to relational, common and some math functions 
+[c3a40760] [c5c7ac2e] [d8715418]
+- Implemented [`sycl_ext_oneapi_annotated_ptr`](doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc) extension [3250db55]
+- Implemented [`sycl_ext_oneapi_cuda_tex_cache_read`](doc/extensions/experimental/sycl_ext_oneapi_cuda_tex_cache_read.asciidoc) extension [5360825e650b]
+- Implemented `sycl::stream`'s size functions definition for device [368b6605]
+- Introduced an API to get thread ID and subdevice ID for ESIMD extension 
+[dd6145e8]
+- Implemented `kernel_bundle::native_specialization_constant()` [4a412307]
+- Implemented SYCL 2020 identities without a known or specified identity 
+[648845fb] [1404ff1d]
+- Implemented basic functionality for the following group types: `ballot_group`, 
+`cluster_group`, `tangle_group`, `opportunistic_group` as part of 
+[`sycl_ext_oneapi_non_uniform_groups`](doc/extensions/proposed/sycl_ext_oneapi_non_uniform_groups.asciidoc) extension [86894205]
+- Implemented `is_group` trait and changed the definition of `is_group_v` to be 
+in line with SYCL 2020 [10e07a38]
+- Implemented `atomic_memory_scope_capabilities` and 
+`atomic_memory_order_capabilities` device queries for OpenCL and Level Zero 
+backends [2816c14072f3] [b18e6ea83c62]
+- Implemented `atomic_fence_scope_capabilities` and 
+`atomic_fence_order_capabilities` device queries for OpenCL and Level Zero 
+backends [632946ac1cbf]
+- Implemented `sycl::ext::intel::experimental::esimd::bfn` function from ESIMD 
+extension [25d0475f]
+- Made `sycl::ext::oneapi::experimental::joint_reduce` work with 
+`sycl::sub_group` [cb91c232]
+
+### Documentation
+- Added [`sycl_ext_oneapi_annotated_ptr`](doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc), [`sycl_ext_intel_fpga_kernel_arg_properties`](doc/extensions/proposed/sycl_ext_intel_fpga_kernel_arg_properties.asciidoc), and 
+[`sycl_ext_oneapi_kernel_arg_properties`](doc/extensions/proposed/sycl_ext_oneapi_kernel_arg_properties.asciidoc) extention specifications [b423e1db]
+- Reintroduced `device_ptr` and `host_ptr` under SYCL namespace for 
+[`sycl_ext_intel_usm_address_spaces`](doc/extensions/supported/sycl_ext_intel_usm_address_spaces.asciidoc) [ea8a0e56]
+- Moved [`sycl_ext_oneapi_memcpy2d`](doc/extensions/supported/sycl_ext_oneapi_memcpy2d.asciidoc) to supported [751acf35]
+- Updated documentation for `-fsycl-no-rdc` mode [a27e6fc5]
+- Added an initial draft of non-uniform sub-groups proposal [cc7cbd28] 
+[a5d62c8a]
+- Added a draft of 16-bit atomics extension [1b973aed]
+- Added an extension specification for USM allocation functions with properties 
+support [3600f922]
+- Added `sycl_ext_oneapi_peer_access` extention specification [b5bce772] 
+[37960cda]
+
+## Improvements
+
+### SYCL Compiler
+- Improved `sycl::reqd_work_group_size` for optional dimensions [b211d0df]
```

Comment:
Tagging @steffenlarsen here as well. If I'm not mistaken, this is not a user-visible change

Reply:
It is somewhat user-visible. Due to index-flipping, certain combinations of arguments would not be handled correctly and we would not handle the attribute correctly. You might argue that it was more a fix than an improvement though.